modules/assignment/src/main/java/com/intuit/wasabi/assignment/impl/AssignmentsImpl.java

```diff
@@ -426,29 +431,69 @@ public Assignment getAssignment(User.ID userID, Application.Name applicationName
+     *                         NULL when called from AssignmentsResource.getBatchAssignments()
+     *                         PRESENT when called from AssignmentsImpl.doPageAssignments()
+     *
+     * @return
+     */
     @Override
     public List<Map> doBatchAssignments(User.ID userID, Application.Name applicationName, Context context,
                                             boolean createAssignment, boolean forceInExperiment, HttpHeaders headers,
                                             ExperimentBatch experimentBatch, Page.Name pageName,
                                             Map<Experiment.ID, Boolean> allowAssignments) {
 
-        // Get the metadata of all the experiments for this application
-        Table<Experiment.ID, Experiment.Label, Experiment> allExperiments = repository.getExperimentList(applicationName);
+        long sTime,totalSTime;
+        totalSTime=System.currentTimeMillis();
+        sTime=System.currentTimeMillis();
 
-        List<Map> allAssignments = new ArrayList<>();
+        //allowAssignments is NULL when called from AssignmentsResource.getBatchAssignments()
+        Set<Experiment.ID> experimentIds = allowAssignments==null?null:allowAssignments.keySet();
+        PrioritizedExperimentList appPriorities = new PrioritizedExperimentList();
+        Map<Experiment.ID, com.intuit.wasabi.experimentobjects.Experiment> experimentMap = new HashMap<>();
+        Table<Experiment.ID, Experiment.Label, String> userAssignments = HashBasedTable.create();
+        Map<Experiment.ID, BucketList> bucketMap = new HashMap<>();
+        Map<Experiment.ID, List<Experiment.ID>> exclusionMap = new HashMap<>();
 
-        // Get the assignments for userID across all experiments in applicationName for the context
-        Table<Experiment.ID, Experiment.Label, String> userAssignments =
-                assignmentsRepository.getAssignments(userID, applicationName, context, allExperiments);
-        PrioritizedExperimentList appPriorities = priorities.getPriorities(applicationName, false);
-        Set<Experiment.ID> experimentSet = allExperiments.rowKeySet();
-        Map<Experiment.ID, BucketList> bucketList = getBucketList(experimentSet);
-        Map<Experiment.ID, List<Experiment.ID>> exclusives = getExclusivesList(experimentSet);
+        //Populate required experiment metadata along with all the existing user assignments for the given application
+        assignmentsRepository.populateExperimentMetadata(userID, applicationName, context, experimentBatch, experimentIds, appPriorities, experimentMap, userAssignments, bucketMap, exclusionMap);
 
+        List<Map> allAssignments = new ArrayList<>();
         // iterate over all experiments in the application in priority order
         for (PrioritizedExperiment experiment : appPriorities.getPrioritizedExperiments()) {
+            if(LOGGER.isDebugEnabled()) LOGGER.debug("Now processing: {}", experiment);
```

Comment:
Added this as having isDebugEnabled avoids a string concatenation in the case if logging LEVEL is not debug...